libobs/obs-encoder.h

```diff
@@ -37,6 +37,7 @@ typedef struct obs_encoder obs_encoder_t;
 #define OBS_ENCODER_CAP_DYN_BITRATE (1 << 2)
 #define OBS_ENCODER_CAP_INTERNAL (1 << 3)
 #define OBS_ENCODER_CAP_ROI (1 << 4)
+#define OBS_ENCODER_CAP_SELF_SCALE (1 << 5)
```

Comment:
As this is an encoder cap, we can probably just name it `OBS_ENCODER_CAP_SCALING`. Documentation can cover what it does exactly.